CONTRIBUTING.md

```diff
@@ -123,6 +120,24 @@ To add a test to the Flutter package, create a file whose name
 ends with `_test.dart` in the `packages/flutter/test` directory. The
 test should have a `main` function and use the `test` package.
 
+Working with flutter tools
+--------------------------
+
+The flutter shell tools itself is built when you run `flutter` for the first time and each time
+you run `flutter upgrade`. If you want to alter and re-test the shell tool's behavior itself,
+delete `bin/cache/flutter_tools.*` so that the shell tool will be rebuilt from Dart sources
```

Comment:
Rather than recommending deleting the cache I would recommend committing the change to your local repo.